tests/framework/extensions/ingresses/ingresses.go

```diff
@@ -1,21 +1,62 @@
 package ingresses
 
 import (
+	"fmt"
+	"io"
+	"net/http"
+
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
-	"github.com/rancher/rancher/tests/framework/extensions/charts"
 )
 
 const (
 	IngressSteveType = "networking.k8s.io.ingress"
 )
 
-// AccessIngressExternally checks if the ingress is accessible externally,
+// GetExternalIngressResponse gets a response from a specific hostname and path,
+// it returns the response and an error if any.
+func GetExternalIngressResponse(client *rancher.Client, hostname string, path string, isWithTLS bool) (*http.Response, error) {
+	protocol := "http"
+
+	if isWithTLS {
+		protocol = "https"
+	}
+
+	url := fmt.Sprintf("%s://%s/%s", protocol, hostname, path)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+client.RancherConfig.AdminToken)
+
+	resp, err := client.Management.APIBaseClient.Ops.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return resp, nil
+}
+
+// ConvertHTTPBodyToString converts the body of an http response to a string
+func ConvertHTTPBodyToString(resp *http.Response) (string, error) {
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	bodyString := string(bodyBytes)
+	return bodyString, nil
+}
```

Comment:
I understand making this public, but I think we can append L44 to L50 to `GetExternalIngressResponse`. This seems a bit more generic for this package. We want to limit this because we usually would want to put HTTP-related things to the clients rather than extensions.

Reply:
agreed. If we want to export a function like this, it doesn't belong here.